File: ExamHallBooking.DataAccess/Data/Models/Appointment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ExamHallBooking.DataAccess.Data.Models
{
    public class Appointment
    {
        [Key]
        public int ID { get; set; }

        [MaxLength(150), Column(TypeName = "nvarchar(150)")]
        public string ExamHall { get; set; } = "Lecture hall";

        [MaxLength(300), Column(TypeName = "nvarchar(300)")]
        public string LectureName { get; set; } = "Lecture Name";

        public int NumOfStudent { get; set; } = 100;

        public int Year { get; set; } = 4;
        public int Semester { get; set; } = 7;

        public string Subject { get; set; } = "EC1020";

        public DateTime CreatedDate { get; set; } = DateTime.Now;
        public DateTime ModifiedDate { get; set; } = DateTime.Now;

        public DateTime Date { get; set;} = DateTime.Now;

        [MaxLength(100), Column(TypeName = "nvarchar(100)")]
        public string AcademicStaff { get; set; } = "academicStaff";

        [MaxLength(10), Column(TypeName ="nvarchar(10)")]
        public string Time { get; set; } = "12:30";

        public string EndTime { get; set; } = "01:30";

        public bool Done { get; set; } = false;
        public bool Deleted { get; set; } = false;

        public byte LevelOfImportance { get; set; } = 2;
    }
}

File: ExamHallBooking.Presentation/ClientApp/src/components/ComAdmin/Appointment.js
import { activeId, entry, openModal } from "./Lib";


export default function Appointment(props) {

    const handlingDelete = (id) => {
        activeId.id = id
        // update state
        props.stateListener(Math.random() * 848 * Math.random())
        //open edit modal
        openModal("delete-modal")
    }
    const handlingDeleteP = (id) => {
        activeId.id = id
        // update state
        props.stateListener(Math.random() * 848 * Math.random())
        //open edit modal
        openModal("deleteP-modal")
    }
    const handlingEdit = (row) => {
        Object.assign(entry, row)
        // update state
        props.stateListener(Math.random() * 548 * Math.random())
        //open edit modal
        openModal("edit-modal")
    }

    const levelOfImportance = ["End Exam", "Mid Exam", "Quiz", "Assignmnet", "", ""];
    const examHall = ["Hall 01 (Admin)", "Hall 02 (Computer)"]
    return (
        <tr className={props.item.deleted ? 'bc-red2' : props.item.done ? 'bc-green2' : ''}>
            <td>{props.item.id}</td>
            <td>{props.item.examHall}</td>
            <td>{props.item.lectureName}</td>
            <td>{props.item.numOfStudent}</td>
            <td>{props.item.year}</td>
            <td>{props.item.semester}</td>
            <td>{props.item.subject}</td>
            <td className={props.item.levelOfImportance === 1 ? 'bc-green' :
                props.item.levelOfImportance === 2 ? 'bc-gold' : props.item.levelOfImportance === 0 ? 'bc-red' : props.item.levelOfImportance === 3 ? 'bc-blue' : ''}>
                {levelOfImportance[props.item.levelOfImportance]}</td>
            <td>{props.item.date.split("T")[0]}</td>
            <td>{props.item.time}</td>
            <td>{props.item.endTime}</td>
            <td>{props.item.academicStaff}</td>
            <td><div className="column edit">
                <div className="btn2 edit" onClick={() => handlingEdit(props.item)}>Edit</div>
            </div></td>
            <td> <div className={`column delete  ${props.item.deleted ? ' not-allowed' : ''}`}>
                <div className={`btn2 delete ${props.item.deleted ? ' no-event' : ''}`} onClick={() => handlingDelete(props.item.id)}>Reject</div>
            </div></td>



        </tr>


    )
}